Tests/ProfilerTest.php

```diff
@@ -87,14 +87,10 @@ public function testRender() : void
             'queries' => $this->logger->queries,
         ]);
 
-        $output = str_replace(["\e[37m", "\e[0m", "\e[32;1m", "\e[34;1m"], '', $output);
-        $this->assertContains("SELECT * FROM foo WHERE bar IN ('foo', 'bar') AND \"\" >= \"\";", $output);
-
-        $expectedEscapedSql = 'SELECT&#x20;&#x0A;&#x20;&#x20;&#x2A;&#x20;&#x0A;FROM&#x20;&#x0A;&#x20;&#x20;foo&#x20;&#x0A;WHERE&#x20;&#x0A;&#x20;&#x20;bar&#x20;IN&#x20;&#x28;&#x3F;,&#x20;&#x3F;&#x29;&#x20;&#x0A;&#x20;&#x20;AND&#x20;&quot;&quot;&#x20;&gt;&#x3D;&#x20;&quot;&quot;';
-        $this->assertContains($expectedEscapedSql, $output);
-        $this->assertSame(
-            "SELECT \n  * \nFROM \n  foo \nWHERE \n  bar IN (?, ?) \n  AND \"\" >= \"\"",
-            html_entity_decode($expectedEscapedSql)
-        );
+        $this->assertSame(1, preg_match('/' . str_replace(
```

Comment:
Those assertions you removed before were important, they were testing escaping functionality that was broken at one point. Important to watch for regressions. Perhaps we should go with true flag as we used to so it's formatted in CLI too after all. Those unit tests are not super important since we have this integration test, but if you want those, you can make assertions for them in similar fashion as was done here.

Reply:
Oh right I think it is indeed messed up… is this assertion  about the `data-clipboard-text`? It looks like this: `data-clipboard-text="&lt;pre&#x20;class&#x3D;&quot;highlight&#x20;highlight-sql&quot;&gt;&lt;span&#x20;class&#x3D;&quot;keyword&quot;&gt;SELECT&lt;&#x2F;span&gt;&#x0A;&#x20;&#x20;&lt;span&#x20;class&#x3D;&quot;symbol&quot;&gt;&#x2A;&lt;&#x2F;span&gt;&#x0A;&lt;span&#x20;class&#x3D;&quot;keyword&quot;&gt;FROM&lt;&#x2F;span&gt;&#x0A;&#x20;&#x20;&lt;span&#x20;class&#x3D;&quot;word&quot;&gt;foo&lt;&#x2F;span&gt;&#x0A;&lt;span&#x20;class&#x3D;&quot;keyword&quot;&gt;WHERE&lt;&#x2F;span&gt;&#x0A;&#x20;&#x20;&lt;span&#x20;class&#x3D;&quot;word&quot;&gt;bar&lt;&#x2F;span&gt;&#x20;&lt;span&#x20;class&#x3D;&quot;keyword&quot;&gt;IN&lt;&#x2F;span&gt;&#x20;&#x28;&lt;span&#x20;class&#x3D;&quot;word&quot;&gt;&#x3F;&lt;&#x2F;span&gt;&lt;span&#x20;class&#x3D;&quot;symbol&quot;&gt;,&lt;&#x2F;span&gt;&#x20;&lt;span&#x20;class&#x3D;&quot;word&quot;&gt;&#x3F;&lt;&#x2F;span&gt;&#x29;&#x0A;&#x20;&#x20;&lt;span&#x20;class&#x3D;&quot;keyword&quot;&gt;AND&lt;&#x2F;span&gt;&#x20;&lt;span&#x20;class&#x3D;&quot;string&quot;&gt;&amp;quot&#x3B;&amp;quot&#x3B;&lt;&#x2F;span&gt;&#x20;&lt;span&#x20;class&#x3D;&quot;symbol&quot;&gt;&amp;gt&#x3B;&lt;&#x2F;span&gt;&lt;span&#x20;class&#x3D;&quot;symbol&quot;&gt;&#x3D;&lt;&#x2F;span&gt;&#x20;&lt;span&#x20;class&#x3D;&quot;string&quot;&gt;&amp;quot&#x3B;&amp;quot&#x3B;&lt;&#x2F;span&gt;&lt;&#x2F;pre&gt;"`

Reply:
I have found an error in this PR that, when fixed, leads to `data-clipboard-text="SELECT&#x20;&#x2A;&#x20;FROM&#x20;foo&#x20;WHERE&#x20;bar&#x20;IN&#x20;&#x28;&#x27;foo&#x27;,&#x20;&#x27;bar&#x27;&#x29;&#x20;AND&#x20;&quot;&quot;&#x20;&gt;&#x3D;&#x20;&quot;&quot;&#x3B;"`

Reply:
It would make sense to use CLI formatting for `data-clipboard-text`, I'm going to try that.

Reply:
Since highlighting is turned off, it shouldn't matter. the only difference here are the linebreaks.

Reply:
Okay I finally understood. The test can stay, but has to be changed because I fixed https://github.com/doctrine/sql-formatter/issues/29

Reply:
Yeah I don't care about highlighting, but having something that tests for escaping regressions. And yeah I think it's about data-clipboard-text

Reply:
I think it is fixed now :) Thanks for spotting this, you rock!